native/core/src/execution/shuffle/shuffle_writer.rs

```diff
@@ -294,7 +291,6 @@ struct ShuffleRepartitioner {
     num_output_partitions: usize,
     runtime: Arc<RuntimeEnv>,
     metrics: ShuffleRepartitionerMetrics,
-    reservation: MemoryReservation,
```

Comment:
This is the main change; removing the memory tracking in `ShuffleRepartitioner` because we already track the memory in each instance of `PartitionedBuffer`.